modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaService.java

```diff
@@ -99,6 +107,10 @@ private <R> CompletableFuture<R> sendToReplica(ClusterNode node, ReplicaRequest
                 assert response instanceof ReplicaResponse : IgniteStringFormatter.format("Unexpected message response [resp={}]",
                         response);
 
+                if (response instanceof TimestampAware) {
+                    clock.update(((TimestampAware) response).timestamp());
```

Comment:
How it's possible to get PrimaryReplicaMissException as an messagingService.invoke result?
```
                } else if (throwable instanceof PrimaryReplicaMissException) {
                    throw (PrimaryReplicaMissException) throwable;
                }
```

Reply:
Yes, seem that it's not need.